application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -0,0 +1,348 @@
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.Permission;
+import net.dv8tion.jda.api.entities.Guild;
+import net.dv8tion.jda.api.entities.Member;
+import net.dv8tion.jda.api.entities.User;
+import net.dv8tion.jda.api.events.interaction.SlashCommandEvent;
+import net.dv8tion.jda.api.interactions.Interaction;
+import net.dv8tion.jda.api.interactions.commands.CommandInteraction;
+import net.dv8tion.jda.api.interactions.commands.OptionMapping;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.SubcommandData;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.togetherjava.tjbot.commands.SlashCommandAdapter;
+import org.togetherjava.tjbot.commands.SlashCommandVisibility;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.db.Database;
+import org.togetherjava.tjbot.db.generated.tables.BanSystem;
+import org.togetherjava.tjbot.db.generated.tables.KickSystem;
+import org.togetherjava.tjbot.db.generated.tables.WarnSystem;
+import org.togetherjava.tjbot.db.generated.tables.records.BanSystemRecord;
+import org.togetherjava.tjbot.db.generated.tables.records.KickSystemRecord;
+import org.togetherjava.tjbot.db.generated.tables.records.WarnSystemRecord;
+
+import java.awt.*;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+import static org.togetherjava.tjbot.commands.utils.MessageUtils.replyEphemeral;
+
+/**
+ * This command can be used to retrieve data for moderation commands. For example this command
+ * allows you to retrieve how many warns a certain user has.
+ */
+public class AuditCommand extends SlashCommandAdapter {
+    private static final Logger logger = LoggerFactory.getLogger(AuditCommand.class);
+    private static final String ACTION_VERB = "audit";
+    private static final String WARN_SUBCOMMAND = "warn";
+    private static final String WARN_USER_OPTION = "user";
+    private static final String WARNED = "warned";
+    private static final String KICK_SUBCOMMAND = "kick";
+    private static final String KICK_USER_OPTION = "user";
+    private static final String KICKED = "kicked";
+    private static final String BAN_SUBCOMMAND = "ban";
+    private static final String BAN_USER_OPTION = "user";
+    private static final String BANNED = "banned";
+    private static final String DESCRIPTION = "The user who you want to get the values for";
+    private static final String OPTION_IS_NULL = "The option is null";
+    private static final String GUILD_IS_NULL = "he guild is null";
+    private static final String THE_USER = "the user ";
+    private static final String COLOR_CODE = "#895FE8";
+    private final Database database;
+    private final Predicate<String> hasKickANDWarnRequiredRole;
+    private final Predicate<String> hasBanRequiredRole;
```

Comment:
Dont require individual roles. Just `hasRequiredRole` and let it be the HEAVY moderation role and thats it.